system/Database/BaseConnection.php

```diff
@@ -1201,7 +1203,7 @@ public function escape($str)
         }
 
         if (is_numeric($str) && $str < 0) {
```

Comment:
Why is there a need to add this for negative numbers? I think we can remove this if block entirely as the last return will do the job.

Reply:
And it seems inconsistent, with negative numbers we get a string. For positive, we can numbers.

Reply:
Good point!

For now, I remove it.

Reply:
However, I am not sure why this commit https://github.com/codeigniter4/CodeIgniter4/commit/19c173a49f35388d90736b6ee596f33a5f29a870 was fixed #606.

Reply:
let's have @lonnieezell shed the light on this. 😂

Reply:
I don't remember exactly. It's been 4 years :) . Looking back through the issue, though, I'm guessing that wrapping it in quotes would get it past the escaping and adding of the slash, which was then converted by MySQL into an INT or whatever type when it was saved to the table. 

I would say as long as we have tests in place that ensure his problem is still fixed, and it doesn't escape the negative sign, this is fine.

Reply:
@lonnieezell Thanks!